tests/coverage/branch.coverage

```diff
@@ -0,0 +1,120 @@
+   LL|       |#![feature(coverage_attribute)]
+   LL|       |// edition: 2021
+   LL|       |// compile-flags: -Zunstable-options -Cinstrument-coverage=branch
+   LL|       |
+   LL|       |macro_rules! do_not_merge_with_signature {
+   LL|       |    () => { if true {} else {} };
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+   LL|       |}
+   LL|       |
+   LL|     15|fn if_not(cond: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|       |        !
+   LL|     15|        cond
+  ------------------
+  |  Branch (LL:9): [True: 10, False: 5]
+  ------------------
+   LL|      5|    {
+   LL|      5|        println!("cond was false");
+   LL|     10|    } else {
+   LL|     10|        println!("cond was true");
+   LL|     10|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_and(a: bool, b: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|     15|        a
+  ------------------
+  |  Branch (LL:9): [True: 12, False: 3]
+  ------------------
+   LL|       |        &&
+   LL|     12|        b
+  ------------------
+  |  Branch (LL:9): [True: 8, False: 4]
+  ------------------
+   LL|      8|    {
+   LL|      8|        println!("both");
+   LL|      8|    } else {
+   LL|      7|        println!("not both");
+   LL|      7|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_or(a: bool, b: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|     15|        a
+  ------------------
+  |  Branch (LL:9): [True: 12, False: 3]
+  ------------------
+   LL|       |        ||
+   LL|      3|        b
+  ------------------
+  |  Branch (LL:9): [True: 14, False: 1]
+  ------------------
+   LL|     14|    {
+   LL|     14|        println!("either");
+   LL|     14|    } else {
+   LL|      1|        println!("neither");
+   LL|      1|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_let(cond: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|       |        let
+   LL|       |        true
+   LL|       |        =
+   LL|     15|        cond
```

Comment:
This currently doesn't get branch coverage at all. I'd like to fix that if possible, but for me it's a lower priority than getting the logical ops correct.